charmcraft/env.py

```diff
@@ -22,6 +22,13 @@
 import pathlib
 from typing import Optional
 
+from charmcraft.const import DEFAULT_SHARED_CACHE_DIR, SHARED_CACHE_ENV_VAR
+
+
+def get_host_shared_cache_path():
+    """Path for host shared cache."""
+    return pathlib.Path(os.getenv(SHARED_CACHE_ENV_VAR, DEFAULT_SHARED_CACHE_DIR)).expanduser()
```

Comment:
Please use the system shared cache directory and ensure the cache directory exists.

We already have `platformdirs` as an indirect dependency, so we can make it a direct dependency without much harm. Platformdirs has a [user_cache_path](https://platformdirs.readthedocs.io/en/latest/api.html#cache-directory) that can be filled in like:

```suggestion
    shared_cache_env = os.getenv(SHARED_CACHE_ENV_VAR)
    if shared_cache_env is not None:
        cache_path = pathlib.Path(shared_cache).expanduser().resolve()
        cache_path.mkdir(parents=True, exist_ok=True)
        return cache_path
    return platformdirs.user_cache_path(appname="charmcraft", ensure_exists=True)
```